src/hotspot/share/runtime/safepoint.cpp

```diff
@@ -379,7 +379,7 @@ void SafepointSynchronize::begin() {
   if (SafepointTimeout) {
     // Set the limit time, so that it can be compared to see if this has taken
     // too long to complete.
-    safepoint_limit_time = SafepointTracing::start_of_safepoint() + (jlong)SafepointTimeoutDelay * (NANOUNITS / MILLIUNITS);
+    safepoint_limit_time = SafepointTracing::start_of_safepoint() + (jlong)SafepointTimeoutDelay * NANOSECS_PER_MILLISEC;
```

Comment:
I think it should be `(SafepointTimeoutDelay * NANOSECS_PER_MILLISEC)` before converting to jlong.

Reply:
Thank you @TheRealMDoerr for reviewing. I just committed a change with your suggestion.